handlers/life_journey.go

```diff
@@ -28,22 +30,97 @@ func (h *Handler) LifeJourney(c echo.Context) error {
+	lifeJourneyList, err := getLifeJourneyBenefitIds(lifeJourneyRequest.Id)
 	if err != nil {
 		c.Logger().Error(err)
 		return c.JSON(http.StatusBadRequest, lifeJourneyResponse)
 	}
 
-	if lifeJourneyRequest.Id != "" {
+	lifeJourneyResponse.LifeJourneyList = lifeJourneyList
+	return c.JSON(http.StatusOK, lifeJourneyResponse)
+}
+
+// LifeJourneyBenefits
+// @Summary Request Info on Life Journey Related Benefits
+// @ID life-journey-benefits
+// @Accept  json
+// @Produce json
+// @Success 200 {object} []models.Benefits
+// @Failure 400 {object} renderings.BenefitServiceError
+// @Failure 404 {object} renderings.BenefitServiceError
+// @Failure 500 {object} renderings.BenefitServiceError
+// @Router /lifejourneys/:id/benefits [get]
+func (h *Handler) LifeJourneyBenefits(c echo.Context) error {
+	var lifeJourneyBenefitsResponse []models.Benefits
+	lifeJourneyBenefitsRequest := new(bindings.LifeJourneyBenefitsRequest)
+
+	// Bind the request into our request struct
+	if err := c.Bind(lifeJourneyBenefitsRequest); err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, lifeJourneyBenefitsResponse)
+	}
+
+	if lifeJourneyBenefitsRequest.Id == "" {
+		return c.JSON(http.StatusUnprocessableEntity, lifeJourneyBenefitsResponse)
+	}
+
+	var lifeJourneyList []models.LifeJourney
+	lifeJourneyList, err := getLifeJourneyBenefitIds(lifeJourneyBenefitsRequest.Id)
+	if err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, lifeJourneyBenefitsResponse)
+	}
+
+	if len(lifeJourneyList) != 1 {
+		return c.JSON(http.StatusUnprocessableEntity, lifeJourneyBenefitsResponse)
+	}
+
+	lifeJourneyBenefitsResponse, err = getBenefitsByIds(lifeJourneyList[0].RelatedBenefits)
+	if err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, lifeJourneyBenefitsResponse)
+	}
+
+	return c.JSON(http.StatusOK, lifeJourneyBenefitsResponse)
+}
+
+func getLifeJourneyBenefitIds(id string) (lifeJourneyList []models.LifeJourney, err error) {
+
+	ljList, err := lifejourneys.LifeJourneyService.LoadLifeJourneys()
+	if err != nil {
+		return lifeJourneyList, err
+	}
+
+	if id != "" {
 		for _, lj := range ljList {
-			if lj.ID == lifeJourneyRequest.Id {
-				lifeJourneyResponse.LifeJourneyList = append(lifeJourneyResponse.LifeJourneyList, lj)
+			if lj.ID == id {
+				lifeJourneyList = append(lifeJourneyList, lj)
 				break
 			}
 
 		}
 	} else {
-		lifeJourneyResponse.LifeJourneyList = ljList
+		lifeJourneyList = ljList
 	}
-	return c.JSON(http.StatusOK, lifeJourneyResponse)
+	return lifeJourneyList, nil
+}
+
+func getBenefitsByIds(benefitIds []string) (benefitsList []models.Benefits, err error) {
+
+	benList, err := benefits.BenefitsService.LoadBenefits()
+	if err != nil {
+		return benefitsList, err
+	}
+
+	for _, benId := range benefitIds {
+		for _, ben := range benList {
+			if ben.ID == benId {
+				benefitsList = append(benefitsList, ben)
+				break
+			}
+		}
+	}
+	return benefitsList, nil
 }
```

Comment:
I'm not sure where these two functions should live. They are reusable components but not sure if it is realistic

Reply:
getBenefitsByIds should probably go into the hendler/benefits.go file since it's getting benefits. I would also consider adding a function in the BenefitService that returns a benefit by it's id, then iterating over the ids and calling that method, rather than loading all benefits and iterating over that.

Reply:
I'm going to do a rework and move this code around in a separate PR